internal/controllers/cluster/cluster_controller.go

```diff
@@ -87,6 +87,10 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	if r.Client == nil || r.APIReader == nil || r.ClusterCache == nil || r.RemoteConnectionGracePeriod == time.Duration(0) {
```

Comment:
@sbueringer we spoke about having at least 180s for RemoteConnectionGracePeriod. Is this a good chance to check for this here or was this a different thing?

Reply:
I'll do this as part of my PR and I'm going to enforce it directly on the flag